### PR TITLE
more proxy tests and fixes

### PR DESCRIPTION
**Pull request by jrfastab**
The previous pull request fixed up the build error this pull request adds additional tests and then fixes the code to handle the additional cases.

Note I made a design decision here to drop the local/global CT distinction. The caveat is in the local case we now use an 'IP address' worth of extra memory in the local CT case. The result is the code is much simpler and easier to read.

Thanks!

### CODE REVIEW

**Comment by jrfastab** [comment]
At this point conntrack_local set via 'cilium config conntrack_local={false|true}'  no longer appears to impact anything. If we want to remove it then the attached test case modifications are not needed and a couple additional deletes from endpoint.go could be added.

**Comment by borkmann** [comment]
Still need to change tuples on the Go side. See 9cf0ee258ba9. Can you elaborate on `no longer impact anything`? Wouldn't a switch to local still be a separate map as opposed the commonly shared map for global.

**Comment by jrfastab** [comment]
OK, I need a few more fixes on the go side like @borkmann notes.

File: tests/14-proxy-ingress.sh [hunk 1]
```diff
@@ -16,26 +16,27 @@ CLIENT_LABEL="id.client"
 cleanup
 logs_clear
 
-docker network inspect $TEST_NET 2> /dev/null || {
```
Comment (tgraf) [inline tests/14-proxy-ingress.sh:19]: Can we merge `10-proxy.sh` and `14-proxy-ingress.sh` together?

Comment (jrfastab) [reply tests/14-proxy-ingress.sh:19]: sure. patches collapse easy enough.

Comment (jrfastab) [reply tests/14-proxy-ingress.sh:19]: aha you are not worried about the patches you mean the actual *.sh files, sure I can do that. Follow up PR though.

**Review by borkmann — APPROVED** [review]
LGTM

**Pull request merged**